Common/include/option_structure.hpp

```diff
@@ -2331,6 +2368,70 @@ class COptionBase {
+
+    const bool sst_1994 = std::find(SST_Options, sst_options_end, SST_OPTIONS::V1994) != sst_options_end ||
+                          std::find(SST_Options, sst_options_end, SST_OPTIONS::V1994m) != sst_options_end;
+    const bool sst_2003 = std::find(SST_Options, sst_options_end, SST_OPTIONS::V2003) != sst_options_end ||
+                          std::find(SST_Options, sst_options_end, SST_OPTIONS::V2003m) != sst_options_end;
+ 
+    /* when V2003m or V1994m is selected, we automatically select sst_m as well */
+    bool sst_m     = std::find(SST_Options, sst_options_end, SST_OPTIONS::MODIFIED) != sst_options_end ||
+                     std::find(SST_Options, sst_options_end, SST_OPTIONS::V1994m) != sst_options_end ||
+                     std::find(SST_Options, sst_options_end, SST_OPTIONS::V2003m) != sst_options_end;
+
+    const bool sst_sust  = std::find(SST_Options, sst_options_end, SST_OPTIONS::SUST) != sst_options_end;
+    const bool sst_v     = std::find(SST_Options, sst_options_end, SST_OPTIONS::VORTICITY) != sst_options_end;
+    const bool sst_kl    = std::find(SST_Options, sst_options_end, SST_OPTIONS::KL) != sst_options_end;
+    const bool sst_uq    = std::find(SST_Options, sst_options_end, SST_OPTIONS::UQ) != sst_options_end;
+
+    // Parse base version
+    if (sst_1994 && sst_2003) {
+      SU2_MPI::Error("Two versions (1994 and 2003) selected for SST Options. Please choose only one.", CURRENT_FUNCTION);
+    } else if (sst_2003) {
+      /*---  sst-2003m model is sst2003 + sstm ---*/
+      if(rank==MASTER_NODE) cout << "Note: currently only the SST-2003m (modified) has been implemented. " << endl;
+      SSTParsedOptions.version = SST_OPTIONS::V2003;
+      sst_m = true;
+    } else if (sst_1994){
+      /* --- Add warning to switch to SST-2003m --- */
+      if(rank==MASTER_NODE) 
+        cout << "warning: the current SST-1994m model is inconsistent with literature. We recommend to use the SST-2003m model." << endl
+             << "In the future, the 2003m model will become the default SST model. " << endl;
+      SSTParsedOptions.version = SST_OPTIONS::V1994;
+      sst_m = true;
+    } else {
+      /* use the most recent model as the default*/
+      if (rank==MASTER_NODE) cout << "SST_OPTIONS not found! using default SST-V1994m. " << endl;
+      SSTParsedOptions.version = SST_OPTIONS::V1994;
+      sst_m = true;
+    }
+
+    // Parse production modifications
+    if ((sst_v + sst_kl + sst_uq) > 1) {
+      SU2_MPI::Error("Please select only one SST production term modifier (V, KL, UQ).", CURRENT_FUNCTION);
+    } else if (sst_v) {
+      SSTParsedOptions.production = SST_OPTIONS::VORTICITY;
+    } else if (sst_kl) {
+      SSTParsedOptions.production = SST_OPTIONS::KL;
+    } else if (sst_uq) {
+      SSTParsedOptions.production = SST_OPTIONS::UQ;
+    }
+
+    // Parse boolean options
+    SSTParsedOptions.sust = sst_sust;
+    SSTParsedOptions.modified = sst_m;
+    SSTParsedOptions.uq = sst_uq;
```

Comment:
is this boolean unneeded since we already set the production enum above?

Reply:
I like the shorthand if(sstParsedOptions.uq) instead of if(sstParsedOptions.production == SST_OPTIONS::UQ).
But I'm fine with removing this boolean if you think it is unnecessary.